tests/unit_tests/hyperion/external_interaction/test_agamemnon.py

```
@@ -0,0 +1,177 @@
+    num_wells, well_width, buffer, expected_width
+):
+    pin = PinType(num_wells, well_width, buffer)
+    assert pin.full_width == expected_width
+
+
+def params_from_loop_type(loop_type: str | None):
+    return {"sample": {"loopType": loop_type}}
+
+
+def test_given_no_loop_type_in_parameters_then_single_pin_returned():
+    assert (
+        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(None))
+        == SinglePin()
+    )
+
+
+@pytest.mark.parametrize(
+    "loop_name, expected_loop",
+    [
+        ("multipin_6x50+9", PinType(6, 50, 9)),
+        ("multipin_6x25.8+8.6", PinType(6, 25.8, 8.6)),
+        ("multipin_9x31+90", PinType(9, 31, 90)),
+    ],
+)
+def test_given_multipin_loop_type_in_parameters_then_expected_pin_returned(
+    loop_name: str, expected_loop: PinType
+):
+    assert (
+        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(loop_name))
+        == expected_loop
+    )
+
+
+@pytest.mark.parametrize(
+    "loop_name",
+    [
+        "nonesense",
+        "single_pin_78x89+1",
+    ],
+)
+@patch("mx_bluesky.hyperion.external_interaction.agamemnon.LOGGER")
+def test_given_completely_unrecognised_loop_type_in_parameters_then_warning_logged_single_pin_returned(
+    mock_logger: MagicMock,
+    loop_name: str,
+):
+    assert (
+        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(loop_name))
+        == SinglePin()
+    )
+    mock_logger.warning.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    "loop_name",
+    [
+        "multipin_67x56",
+        "multipin_90+4",
+    ],
+)
+def test_given_unrecognised_multipin_in_parameters_then_warning_logged_single_pin_returned(
+    loop_name: str,
+):
+    with pytest.raises(Exception) as e:
+        assert (
+            _get_pin_type_from_agamemnon_parameters(params_from_loop_type(loop_name))
+            == SinglePin()
+        )
+        assert "Expected multipin format" in str(e.value)
```

```suggestion
    with pytest.raises(Exception) as e:
        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(loop_name))
    assert "Expected multipin format" in str(e.value)
```
I think it was agreed to throw an exception instead of returning `SinglePin()`, so this assertion should be removed. Moreover, `assert "Expected multipin format" in str(e.value)` is never actually ran, so should be outside the manager.